pkg/controller/synchronize.go

```diff
@@ -130,6 +130,12 @@ func applyDeclaration(declaration *v1alpha1.Cluster) resourcetree.ApplyFn {
 			desiredTreeNode.State = boolToState(declaration.Integrations.ArgoCD)
 		case resourcetree.ResourceNodeTypeUsers:
 			desiredTreeNode.State = boolToState(len(declaration.Users) > 0)
+		case resourcetree.ResourceNodeTypePostgres:
+			desiredTreeNode.State = boolToState(false)
+
+			if declaration.Databases != nil {
```

Comment:
What would that look like? I am not sure I understand how I should be checking earlier?

Reply:
You could for example initialize the array in `NewDefaultCluster()` and add a validation ensuring the is initialized (but could be empty). That would save us from having to check if the list is nil

Reply:
I implemented `Databases` as a struct, so we could put in other types there in the future if needed: https://github.com/oslokommune/okctl/pull/398/files/0e88a00388b8481702443c5a7e1da6fa0eb107e4#diff-11c6c7e74102f66ec75387a08390b4ecbc69c16f00a5dabc5e27dec3b879fa1dR59, if it is non-empty it will serialise an empty struct, but I guess that is fine?

Reply:
Oh right. Hmm. Doesn't really matter. You do you. I couldn't find anything of substance to pick on, so I started nitpicking

Reply:
Nitpicking is the best kind of picking :D